### PR TITLE
Rename User interface and add get user / get service account numeric id calls

### PR DESCRIPTION
**Pull request by MuweiHe**
Since we need `List User` and `Get Service Accounts` in the `User` interface, I'm renaming it back to `User` (I named it LoginRealm as I thought that's all we needed). Added 3 routes we need for cli kafka acl to get the user numeric id. 
Added mocks.

### CODE REVIEW

File: user.go [hunk 1]
```diff
@@ -26,23 +27,54 @@ const (
...
+	return reply.Users, nil
+}
+
+// GetServiceAccounts returns a collection of service account users.
+func (s *UserService) GetServiceAccounts(_ context.Context) ([]*User, error) {
+	reply := new(GetServiceAccountsReply)
+
+	_, err := s.sling.New().Get(SERVICEACCOUNTS).Receive(reply, reply)
+	if err := ReplyErr(reply, err); err != nil {
+		return nil, WrapErr(err, "error listing service account")
```
Comment (sgagniere) [inline user.go:61]:
```suggestion
		return nil, WrapErr(err, "error listing service accounts")
```

**Review by sgagniere — APPROVED** [review]
lgtm

File: interfaces.go [hunk 2]
```diff
@@ -40,7 +40,11 @@ type Growth interface {
 	GetFreeTrialInfo(context.Context, int32) ([]*GrowthPromoCodeClaim, error)
 }
 
-type LoginRealm interface {
+// User service allows managing users in Confluent Cloud
+type UserInterface interface {
+	List(context.Context) ([]*User, error)
+	GetServiceAccounts(context.Context) ([]*User, error)
+	GetServiceAccount(context.Context, int32) (*User, error)
 	LoginRealm(context.Context, *GetLoginRealmRequest) (*GetLoginRealmReply, error)
```
Comment (brianstrauch) [inline interfaces.go:48]: It seems like we're combining two APIs into one here? Should the user management / login realm functions really be grouped together?

Comment (MuweiHe) [reply interfaces.go:48]: They were together...  I renamed it to `loginRealm` as I thought no other `User` methods would be needed.

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**